### PR TITLE
Feature/various UI improvements

### PR DESCRIPTION
**Pull request by Dassderdie**
Fixes:
* #434 (Autosaving is now debounced -> fewer actions; and doesn't propose invalid actions)
* #433 
* Always show transfer point dropdown

### CODE REVIEW

File: frontend/src/app/pages/exercises/exercise/shared/exercise-map/feature-managers/transfer-point-feature-manager.ts [hunk 1]
```diff
@@ -85,79 +85,7 @@ export class TransferPointFeatureManager extends ElementFeatureManager<TransferP
...
-            if (
-                reachableTransferPointIds.length === 0 &&
-                reachableHospitalIds.length === 1
-            ) {
-                proposeTransfer(reachableHospitalIds[0], 'hospital');
-                return true;
-            }
-        }
-
+        // We always want to show the chooseTransferPopup to
```
Comment (Dassderdie) [inline frontend/src/app/pages/exercises/exercise/shared/exercise-map/feature-managers/transfer-point-feature-manager.ts:88]:
```suggestion
        // Always show the chooseTransferPopup
```

File: shared/src/store/action-reducers/transfer-point.ts [hunk 2]
```diff
@@ -141,8 +143,13 @@ export namespace TransferPointActionReducers {
                     'transferPoints',
                     transferPointId
                 );
-                transferPoint.internalName = internalName;
-                transferPoint.externalName = externalName;
+                // Empty strings are ignored
```
Comment (ClFeSc) [inline shared/src/store/action-reducers/transfer-point.ts:146]: This is a breaking change. Could this break something? Probably not. Do we want to risk it?

File: frontend/src/app/pages/exercises/exercise/shared/exercise-map/shared/transfer-point-popup/transfer-point-popup.component.html [hunk 3]
```diff
@@ -185,8 +241,6 @@ <h5 class="popover-header">
                     </div>
```
Comment (ClFeSc) [inline frontend/src/app/pages/exercises/exercise/shared/exercise-map/shared/transfer-point-popup/transfer-point-popup.component.html:241]: Making the comment here as GitHub doesn't allow comments on all lines: It's probably not new, but I just noticed that the transfer time to the hospital in the transfer point popup is slightly offset to the top. Maybe we can fix this here?

**Review by ClFeSc — COMMENTED** [review]
Just very little things. Nice work!

File: frontend/src/app/pages/exercises/exercise/shared/exercise-map/shared/transfer-point-popup/transfer-point-popup.component.html (hunk 3 above)
Comment (Dassderdie) [reply frontend/src/app/pages/exercises/exercise/shared/exercise-map/shared/transfer-point-popup/transfer-point-popup.component.html:241]: Current:
image
No monospace
image
Both monospace
image

The offset is due to the font being monospace. Without monospace different numbers above each other don't look good. -> I let it be like it is now.

**Review by ClFeSc — APPROVED** [review]

**Review by ClFeSc — APPROVED** [review]

**Comment by Dassderdie** [comment]
@ClFeSc I made the pretriage table smaller and removed one unnecessary check in a function. 

| Before | After |
| --- | --- |
| image | image |

I don't think this requires another review.

**Pull request merged**